File: frontend/src/app/models/order.ts
export interface OrderRequest {
  userId: number
  items: OrderItem[]
  shippingAddress: string
  phoneNumber: string
  totalAmount: number
}

export interface OrderItem {
  backpackId: number
  quantity: number
  price: number
}
export interface Order {
  id: number;
  userId: number;
  user?: {
    userName: string;
    email: string;
  };
  orderDate: string;
  totalPrice: number;
  status: string;
  shippingAddress: string;
}

export interface OrderResponse {
  id: number
  message: string
}

File: backend/BackpackStoreFS/Models/Entities/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BackpackStoreFS.Models.Entities
{
    public class Order
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [ForeignKey("UserId")]
        public User User { get; set; } = null!;

        public DateTime OrderDate { get; set; } = DateTime.UtcNow;

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalPrice { get; set; }

        [Required]
        [StringLength(20)]
        public string Status { get; set; } = "Pending";

        [Required]
        [StringLength(500)]
        public string ShippingAddress { get; set; } = null!;

        [StringLength(100)]
        public string? TrackingNumber { get; set; }

        public List<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}
